### PR TITLE
Feature Freeze

### PR DESCRIPTION
**Pull request by Cyberboss**
/tg/station 13 is currently in a feature freeze. All PRs opened after this one are subject to the freeze rules until it ends.

#### What is allowed:
Fixes
Code Improvement
Tool Modifications
Sprite Edits
Sound Changes
Map Edits that fix issues

#### What is not allowed:
Any kind of feature
Any kind of tweak/balance change
Map edits with the intent of modifying topology
Feature removals
Reverts that don't fix immediate issues

The refactors are on approval, ask in #coderbus before starting coding to get said approval

### CODE REVIEW

**Comment by Tlaltecuhtli** [comment]
first

**Comment by davethwave** [comment]
Well it's time to stock up for the winter...

**Comment by MrStonedOne** [comment]
\>the list is shuffled internally

Forgets about the fact that the job with the most motivation to attempt to hack anything only starts the round with one or two accesses.

Its comical and thats why I got the headmins to agree that this is a OP feature and not a in game exploit. Powergaming rules may apply to the excessive use of this but that's it.

But the fact remains it is Over Powered for how little effort it takes and the funny bit is its easier to exploit the less access you have.

**Comment by Xhuis** [comment]
My PRs! De GBP does nothing!

**Comment by Robustin** [comment]
>Putting the finishing touches on a Ruin PR with my directional flashlight
>ITS NOW MIDNIGHT IN IRAQ, SHUT IT DOWN

This place constantly places my love for this game at odds with the shitshow I get dragged through every time I rely on someone in charge here to act like an adult.

**Comment by Iamgoofball** [comment]
This is why you open WIPs

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1,70 +1 @@
...
-# ShizCalev
-
-/_maps/ @ShizCalev
-/sound/ @ShizCalev
-
-# Multiple Owners
-
-/code/controllers/subsystem/air.dm @duncathan @MrStonedOne
-/icons/ @ShizCalev @WJohn
+/ @JohnOxford
```
Comment (Iamgoofball) [inline .github/CODEOWNERS]: *gavel smack*

**Review by Iamgoofball — CHANGES_REQUESTED** [review]
Racism/10

**Comment by Jordie0608** [comment]
There doesn't seem to be an end condition, is it still being mulled over?

**Comment by KorPhaeron** [comment]
@Jordie0608 

Yeah. Was just going to go for a month since issue target counts are hard to estimate properly with all the dupe or old issues we end up closing straight away. Open to better ideas though.

**Comment by Buggy123** [comment]
Oranges, the feature freeze is set for March 21, this was posted at about 4pm EST. That is not even 12:01 AM in any of America's territories. Nor in any of the EU.

For reference,  here is a approximate map of all of the world where it was NOT March 21:
great time
And the part where it WAS:
bad time


So, it was the **absolute earliest time that this could possibly be justified** in Russia, China, India, and Austraila. The latest this was posted _anywhere in the world_ (I.e. New Zealand and the ass-end of Russia) was **5 AM**.

No. Get the fuck out of here. N. O. Complete bullcrap. "BUT IT WAS A REASONABLE TIME IN-" no. Go. Sodomize. Yourself. With. A. Cactus.

IN THE LATEST GODDAMN TIMEZONES IN THE WORLD IT WAS 5 IN THE FUCKING MORNING, 90% OF PEOPLE WEREN'T EVEN AWAKE AT THE LATEST GOD DAMN TIME THIS WAS POSTED. CUT THE BULLSHIT BECAUSE YOU'RE FOOLING NO ONE.


I say we delay this for a day so everyone who wanted to get a last minute PR in can do so.

**Comment by Cyberboss** [comment]
You're right @Buggy123, I should have opened it 5 hours earlier

**Comment by Xhuis** [comment]
This is advanced autism.

**Comment by Robustin** [comment]
@Cyberboss 

I don't actually give a fuck if you start the freeze at 3:33am in whatever fucking time zone manages to have Russia and Laos but skips China entirely - JUST TELL US IN ADVANCE JESUS H CHRIST ITS LIKE ONE THE FUNDAMENTAL THINGS FOR ANY ORGANIZATION THAT ISN'T DELIBERATELY SETTING ITSELF UP FOR FAILURE AS SOME SORT OF FINANCIAL FRAUD SCHEME. WE DONT MAKE MONEY AND I DONT HEAR SPRINGTIME FOR HITLER SO WITH THAT POSSIBILITY OUT JUST PUT ON YOUR BIG BOY PANTS AND GIVE US A FUCKING TIME.

**Comment by optimumtact** [comment]
I didn't say it was a reasonable requiremnet, I'm just laughing at you getting so buttmad over not being able to make feature prs for 30 days out of an entire year lmao, to a github run by volunteers, who dont' have to give you the time of the day if they don't feel like it.

Climb of your entitled high horse because none of the people whining about this change are valued contributors, so if you wanna fuck off and jerk off on some other place, we're not gonna shed a single tear.

**Comment by Buggy123** [comment]
Actually I wasn't planning to make any PRs, I haven't even finished the ones I still have open. I just have zero tolerance for that bullshit.

>Climb of your entitled high horse because none of the people whining about this change are valued contributors

Sorry @Robustin, I know you've submitted almost 300 PRs, including major refactors, gamemode overhauls, balance changes and fixes, but you aren't a valued contributor.

**Comment by optimumtact** [comment]
yes that's right

**Comment by Buggy123** [comment]
Oh, also:

>to a github run by volunteers, who dont' have to give you the time of the day if they don't feel like it.

See, but that's the thing. You're here. Arguing with me. If you didn't care you wouldn't be here, but the fact you've responded to me at all shows that you do.

**Comment by optimumtact** [comment]
also what do you mean by 5am?

**Comment by Jalleo** [comment]
Guys you forhet Jordie made it a tradition so it opens at midnight in Australia. Thats why its always early.

**Comment by Buggy123** [comment]
>also what do you mean by 5am?

Shoot, I think I messed up the conversion.

cropped

EDT is [four hours behind Coordinated Universal Time, aka UTC](https://www.timeanddate.com/time/zones/edt)

[The latest timezone that exists is 12 hours ahead of UTC, and from there it loops back around to 12 hours behind.](https://upload.wikimedia.org/wikipedia/commons/e/e8/Standard_World_Time_Zones.png)

So four hours ahead would be 8pm in UTC, and then 12 more would be 8 AM. Still too early.

Also what the fuck? Where the fuck is 13 hours ahead? That shouldn't be a thing.

**Comment by kevinz000** [comment]
btfo

**Comment by Buggy123** [comment]
"This was made either on the wrong day or early as fuck for a large portion of the civilized world, and was still very early for even the latest timezones in the world" is not blown the fuck out, even if my initial calculation was 4 hours off. It was still the wrong day for the entirety of north+south america, greenland, and some of europe.

Also you should've just given a goddamn timeframe instead of setting _none_ and then doing it at the first possible moment.

**Comment by davethwave** [comment]
Yeah... It was only about 5pm 3/20 when this go released for me

**Comment by Robustin** [comment]
@optimumtact 

I support freezes, I'm fine with shutting down features to fix issues. It all makes sense, I'm honestly a little relieved by it.

What absolutely infuriates me is that I spent my weekend crunching directional lighting so that it would be in top-form for the space ruin I've been working on. Then when I get ready to open the PR for the only place where directional lighting would actually exist - I find out the freeze started at some random unannounced time and I had missed the window by mere minutes.

So don't warp this into me hating the freeze, its just the utter mismanagement and disdain for contributors that pisses me off to no end.

**Comment by davethwave** [comment]
I agree that there could be a bit more communication with this considering the large effect it has here

**Comment by Qbopper** [comment]
@Robustin instead of making a salt thread and flipping the fuck out on github why didn't you just PM maintainers and say "hey, I was making this PR and the freeze suddenly happened, is it cool if you let this one slide" and then they'll do it like the other times this has happened

**Comment by optimumtact** [comment]
because dramatically overreacting to everything is robustin's MO

**Comment by Rinoka** [comment]
>Also what the fuck? Where the fuck is 13 hours ahead? That shouldn't be a thing.

it's oranges

but yes y'all remember this is run by volunteers which means if you had an Urgent Thing you wanted to merge and had done just... ask

noneed

**Comment by optimumtact** [comment]
it's daylight savings yo

**Comment by AlexTheSysop42** [comment]
What's that lawnmower pic from?

**Comment by MrStonedOne** [comment]
@AlexTheSysop42 http://lmgtfy.com/?q=know+that+meme+lawnmower+upset

**Comment by chesse20** [comment]
Merge this PLEASE!!!!

**Comment by AlexTheSysop42** [comment]
@MrStonedOne It's know your meme get it right

**Comment by CPTANT** [comment]
THERE ARE MORE ISSUES THAN WHEN THE FREEZE STARTED

**Comment by Anonmare** [comment]
When will Global Warming stop all this freezing!?

**Comment by Iamgoofball** [comment]
@robustin it's alright man it's winter for poland and France

**Comment by outbools** [comment]
I want to see dynamic stats about all fix PR numbers during all Feature Freezes prior invention of GBP.
Let's actually see if your rituals summoned Fix God once in a damn time.
It just scares most of coders away, that's it.

**Comment by outbools** [comment]
https://api.github.com/repos/tgstation/tgstation/issues?state=closed&labels=Fix
would be great if someone make a graph out of it

**Comment by Rinoka** [comment]
i was fucking with the pyGithub stuff to try and get some reliable stats but also i realized the search is fine for this so HERE WE GO, search terms written uptop

## Issues Closed (is:issue closed)
FREEZE: **316** issues closed.
PRE-FRZ: **183** issues closed.

## Issues Created (is:issue created)
FREEZE: **306** issues created.
PRE-FRZ: **238** issues created.

## Fix PRs Made (is:pr label:fix created)
FREEZE: **256** fix PRs created. (**233** merged, **206** of which were made after the freeze began.)
PRE-FRZ:  **164** fix PRs created. (**132** merged.)

## Code Improvement PRs Made  (is:pr label:"code improvement" created)
FREEZE: **256** fix PRs created. (**233** merged, **206** of which were made after the freeze began.)
PRE-FRZ:  **37** code improvement PRs created. (**33** merged.)

and yes, PR numbers overlap

i was interested in pulling the numbers on how many users were actively creating PRs but also that's a pain and would require having my computer do a for loop over like 20k entries and i love my computer more than that

but there's a pretty clear spike in fix PRs, offset by the fact that the amount of issues reported barely dropped over the 1mo freeze.

**Comment by Iamgoofball** [comment]
after phil quit for CM freezes died

**Comment by KorPhaeron** [comment]
APC destroyed mission accomplished

Thank you for the hard work everyone,, we now return to our regularly scheduled bikeshedding and feature bloat

**Pull request closed without merging**